fili-core/src/main/java/com/yahoo/bard/webservice/data/config/metric/makers/MetricMaker.java

```diff
@@ -149,23 +153,17 @@ protected TemplateDruidQuery getDependentQuery(String name) {
      * @return The merged query
      */
     protected TemplateDruidQuery getMergedQuery(List<String> names) {
-        if (names.size() <= 0) {
-            String message = "At least 1 name is needed to merge aggregations";
-            LOG.error(message);
-            throw new IllegalArgumentException(message);
-        }
-
-        // Get the initial query
-        TemplateDruidQuery query = getDependentQuery(names.get(0));
-
         // Merge in any additional queries
-        for (int i = 1; i < names.size(); i++) {
-            TemplateDruidQuery additionalQuery = getDependentQuery(names.get(i));
-            query = query.merge(additionalQuery);
-        }
-
-        // Return the merged query
-        return query;
+        return names.stream()
+                .map(metrics::get)
+                .map(LogicalMetric::getTemplateDruidQuery)
+                .reduce(TemplateDruidQuery::merge)
+                .orElseGet(() -> {
+                            String message = "At least 1 name is needed to merge aggregations";
+                            LOG.error(message);
+                            throw new IllegalArgumentException(message);
+                        }
```

Comment:
Is this bracket indented correctly? Typically, when the opening of a lambda is on the same line as a method, I put the closing bracket on the same line as the closing parens. i.e.:

```java
  method(() -> {
      doStuf();
  });
```

and the closing bracket is on its own line only if the start of the lambda is on its own line:


```java
method(
          () -> {
              doStuf();
          }
);
```

I don't have a strong opinion either way, I just wanted to know if we had a convention around this.